horovod/mxnet/__init__.py

```diff
@@ -64,16 +66,23 @@ def _do_allreduce(self, index, grad):
                 index_split = split_list(index, self._num_groups)
 
                 for i, (grads, indices) in enumerate(zip(grad_split, index_split)):
-                    grouped_allreduce_(tensors=grads, average=False, name="{}:{}".format(indices[0], indices[-1]), priority=-i,
+                    tensors_compressed, ctxs = zip(*[self._compression.compress(grad) for grad in grads])
+                    grouped_allreduce_(tensors=tensors_compressed, average=False,
+                                       name="{}:{}".format(names[0], names[-1]), priority=-i,
                                        prescale_factor=1.0 / self._gradient_predivide_factor)
+                    grads = [self._compression.decompress(t, ctx) for t, ctx in zip(tensors_compressed, ctxs)]
```

Comment:
This is not quite right. This leaves the final allreduced and uncompressed results in the local list `grads`, but doesn't actually update the real parameter gradients.